frmts/vrt/gdaltileindexdataset.cpp

```diff
@@ -0,0 +1,3642 @@
+                auto psMD = poBand->oMDMD.Serialize();
+                if (psMD)
+                    bHasBandMD = true;
+                aoBandXML.emplace_back(CPLXMLTreeCloser(psMD));
+            }
+            if (bHasBandMD)
+            {
+                for (int i = 1; i <= nBands; ++i)
+                {
+                    auto poBand =
+                        cpl::down_cast<GDALTileIndexBand *>(papoBands[i - 1]);
+
+                    CPLXMLNode *psBand = CPLCreateXMLNode(psRoot, CXT_Element,
+                                                          GTI_XML_BAND_ELEMENT);
+                    CPLAddXMLAttributeAndValue(psBand, GTI_XML_BAND_NUMBER,
+                                               CPLSPrintf("%d", i));
+                    CPLAddXMLAttributeAndValue(
+                        psBand, GTI_XML_BAND_DATATYPE,
+                        GDALGetDataTypeName(poBand->GetRasterDataType()));
+
+                    const char *pszDescription = poBand->GetDescription();
+                    if (pszDescription && pszDescription[0])
+                        CPLSetXMLValue(psBand, GTI_XML_BAND_DESCRIPTION,
+                                       pszDescription);
+
+                    const auto eColorInterp = poBand->GetColorInterpretation();
+                    if (eColorInterp != GCI_Undefined)
+                        CPLSetXMLValue(
+                            psBand, GTI_XML_BAND_COLORINTERP,
+                            GDALGetColorInterpretationName(eColorInterp));
+
+                    if (!std::isnan(poBand->m_dfOffset))
+                        CPLSetXMLValue(psBand, GTI_XML_BAND_OFFSET,
+                                       CPLSPrintf("%.16g", poBand->m_dfOffset));
+
+                    if (!std::isnan(poBand->m_dfScale))
+                        CPLSetXMLValue(psBand, GTI_XML_BAND_SCALE,
+                                       CPLSPrintf("%.16g", poBand->m_dfScale));
+
+                    if (!poBand->m_osUnit.empty())
+                        CPLSetXMLValue(psBand, GTI_XML_BAND_UNITTYPE,
+                                       poBand->m_osUnit.c_str());
+
+                    if (poBand->m_bNoDataValueSet)
+                    {
+                        CPLSetXMLValue(
+                            psBand, GTI_XML_BAND_NODATAVALUE,
+                            VRTSerializeNoData(poBand->m_dfNoDataValue,
+                                               poBand->GetRasterDataType(), 18)
+                                .c_str());
+                    }
+                    if (aoBandXML[i - 1])
+                    {
+                        CPLAddXMLChild(psBand, aoBandXML[i - 1].release());
+                    }
+                }
+            }
+        }
+
+        if (!CPLSerializeXMLTreeToFile(m_psXMLTree.get(), GetDescription()))
+            eErr = CE_Failure;
+    }
+
+    m_oMapSharedSources.clear();
+    m_dfLastMinXFilter = std::numeric_limits<double>::quiet_NaN();
```

Comment:
following comment added:
    // We also clear the cache of opened sources, in case the user would
    // change the content of a source and would want the GTI dataset to see
    // the refreshed content.